GillianCore/engine/GeneralSemantics/General/GInterpreter.ml

```diff
@@ -541,25 +541,19 @@ struct
 
     let eval_expr = make_eval_expr state in
     match lcmd with
-    | AssumeType (x, t) -> (
-        match Val.from_expr (LVar x) with
+    | AssumeType (e, t) -> (
+        match Val.from_expr e with
```

Comment:
Should this not evaluate the expression properly?

Reply:
I'm not sure it should, the goal is to reverse-type the expression itself, and then add whatever types to the typenv, it's not really required to evaluate the expression, and it's a logic command, so Val.from_expr is guaranteed to exist

Reply:
Ah, but I do want to evaluate it, because I want to type the lvar properly, not the pvar

Reply:
FIxed